modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteInfo.java

```diff
@@ -24,32 +24,45 @@
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.index.reindex.BulkByScrollTask;
 
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
 
 import static java.util.Collections.unmodifiableMap;
 import static java.util.Objects.requireNonNull;
+import static org.elasticsearch.common.unit.TimeValue.timeValueMinutes;
 
 public class RemoteInfo implements Writeable {
+    /**
+     * Default socket timeout for requests that don't have one set.
+     */
+    public static final TimeValue DEAULT_SOCKET_TIMEOUT = timeValueMinutes(1);
```

Comment:
why one minute? It seems a very high value. default is 10 seconds in the rest client, which is already not that low.   wouldn't 30 seconds be enough. For sure you want to make reindex work but if a remote cluster is that slow something is quite wrong and I am not sure increasing the timeout like this will make everything work. I would also prefer that people encountering problems would need to increase it, so that they know they have a problem, rather than leaving a crazy high timeout as default. WDYT?